crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -1100,6 +1127,72 @@ export.get_acct_storage_slots_section_ptr
     exec.get_current_account_data_ptr push.ACCT_STORAGE_SLOTS_SECTION_OFFSET add
 end
 
+#! Returns the memory pointer to the native account's storage slots section.
+#!
+#! Inputs:  []
+#! Outputs: [storage_slots_section_ptr]
+#!
+#! Where:
+#! - storage_slots_section_ptr is the memory pointer to the native account's storage slots section.
+export.get_native_account_storage_slots_ptr
+    exec.get_native_account_data_ptr add.ACCT_STORAGE_SLOTS_SECTION_OFFSET
+end
+
+#! Returns the memory pointer to the initial storage slots of the native account.
+#!
+#! Inputs:  []
+#! Outputs: [account_initial_storage_slots_ptr]
+#!
+#! Where:
+#! - account_initial_storage_slots_ptr is the memory pointer to the initial storage slot values.
+export.get_native_account_initial_storage_slots_ptr
+  exec.get_native_account_data_ptr add.ACCT_INITIAL_STORAGE_SLOTS_SECTION_OFFSET
+end
+
+### ACCOUNT DELTA #################################################
+
+#! Increments the delta of the account's nonce.
+#!
+#! Example: If this procedure is called with nonce_delta 3 and again with nonce_delta 2, then the
+#! stored nonce_delta will be 5.
+#!
+#! Inputs:  [nonce_delta]
+#! Outputs: []
+#!
+#! Where:
+#! - nonce_delta is the value by which the nonce changed.
+export.incr_account_nonce_delta
+    exec.get_account_nonce_delta add mem_store.ACCOUNT_DELTA_NONCE_PTR
+end
```

Comment:
As mentioned in a comment elsewhere, I think we could impose a restriction that account delta can be incremented only once per transaction. This could be for a different PR though.